### DISCUSSION TITLE
Racks - Adjusted/Additional "Add Device" Functionality

### DISCUSSION THREAD

**Discussion by adparis99** [netbox-community/netbox#12873] [Ideas]
I'm not sure if this has been discussed before, but recently I've been making more use of racks/elevations, and one thing I'd like to see in the future is for the _add _device__ link on racks to follow the logic of the _add ip _address__ button/link on interfaces. Specifically, the latter links to the _Add a New IP Address_ page and has the option of searching for and assigning an existing IP, in addition to creating a brand new one, as pictured below:

![image](https://github.com/netbox-community/netbox/assets/16410007/d5afbb41-39c0-41d3-97ae-9565fe1a64f8)

When clicking the former, however, only the _Create_ tab is present. I'm by no means a programmer so I'm not sure of the technical requirement, just curious if anyone else thought it might be beneficial. It would be nice to be able to assign devices to racks in the UI without having to manually go to the devices is properties.

**Comment by lingej** [comment]
I would love to see this function!
